### ISSUE TITLE
Make RequestContext required on protocol methods in the event of an ambiguous call site with grow-up method

### ISSUE THREAD

**Issue by annelo-msft** [Azure/azure-sdk-for-net#28123]
When we have an ambiguous call site between protocol method and convenience method, we must rename one of the methods to create a separate method group.  If we're changing the name of the convenience method, we suggest the suffix `Value`.

https://github.com/Azure/azure-sdk-for-net/blob/main/doc/Data%20Plane%20Code%20Generation/DeveloperDrivenEvolution.md#rename-method-if-needed

If we're changing the name of the protocol method, we need something to indicate we're returning a raw response instead of a model type.

We have considered the following:
- `GetResourceJson`
- `GetResourceRaw`
- `GetResourceResponse`
- `GetResourceRawResponse`

Most of these work for GET and LIST operations, but don't work well for Create, Set, or Delete.
We also have a complication that we make the last word on a Pageable operation plural for generated protocol methods.

### Goals

- [ ] Update https://github.com/Azure/azure-sdk-for-net/blob/main/doc/Data%20Plane%20Code%20Generation/DeveloperDrivenEvolution.md#rename-method-if-needed with new guidance
- [x] Update https://github.com/Azure/azure-sdk-for-net/pull/27363 with chosen approach

**Comment by annelo-msft** [comment]
The decision from our 04/21 design meeting is that, in this case, we will not suffix the protocol method, but instead disambiguate the method signatures by making `RequestContext` and any other parameters in the method signature required.

**Comment by ShivangiReja** [comment]
>- Update https://github.com/Azure/azure-sdk-for-net/pull/27363 with chosen approach

@annelo-msft I've already updated https://github.com/Azure/azure-sdk-for-net/pull/27363.

There was an ambiguous call between protocol and convenience method for `GetDataFeeds` method, based on  04/21 design meeting I used the approach below:
* Added a public protocol method in convenience client and made all the parameters required(check [here](https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs#L287-L391)).
* Since the method was already existed, generator didn't generate the protocol method for `GetDataFeeds`.
